src/Data/Inputs.elm

```diff
@@ -50,11 +50,24 @@ type alias CustomCountryMixes =
 
 fromQuery : Db -> Query -> Result String Inputs
 fromQuery db query =
+    let
+        material =
+            db.materials |> Material.findByUuid query.material
+    in
     Ok Inputs
         |> RE.andMap (Ok query.mass)
-        |> RE.andMap (db.materials |> Material.findByUuid query.material)
+        |> RE.andMap material
         |> RE.andMap (db.products |> Product.findById query.product)
-        |> RE.andMap (db.countries |> Country.findByCodes query.countries)
+        |> RE.andMap
+            (material
+                |> Result.andThen
+                    (\{ defaultCountry } ->
+                        query.countries
+                            -- Update the list of countries: the first country (from the material step) is constrained to be the material's default country
+                            |> updateCountryList 0 defaultCountry
+                            |> (\updatedCountries -> Country.findByCodes updatedCountries db.countries)
```

Comment:
We may want to use [List.Extra.setAt](https://package.elm-lang.org/packages/elm-community/list-extra/latest/List-Extra#setAt) instead.

Reply:
Sure thing, I merely reproduced the code that was present before, but indeed that would be a nice shortcut 👍